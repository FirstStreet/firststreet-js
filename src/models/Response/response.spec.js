const BaseModel = require('./index');
const mock = require('../../__mocks__/hurricaneProperty');

let response = null;

describe('mock', () => {
  beforeEach(() => {
    response = new BaseModel(mock);
  });

  it('should initialize a base model', () => {
    expect(response).toMatchSnapshot();
  });

  it('should return a raw response', () => {
    expect(response.raw).toBe(mock);
  });

  it('should return the correct id', () => {
    expect(response.FSID).toBe(mock.FSID);
  });

  it('should return the correct results', () => {
    expect(response.results).toBe(mock.results);
  });

  it('getDataByYear should return the correct results', () => {
    expect(response.getDataByYear(2023).val).toBeDefined();
  });

  it('getDataByFloodID should return the correct results', () => {
    expect(response.getDataByFloodID('c1').val).toBeDefined();
  });

  it('method chaining should return the correct results', () => {
    expect(response.getDataByYear(2023).getDataByFloodID('c1').val).toBeDefined();
  });

  it('getData() should return the correct results', () => {
    const mockedData = {
      type: 'maxdepth',
      unit: 'meters',
      value: 0,
    };
    expect(response.getData('maxdepth', 'c1', 2023)).toEqual(mockedData);
  });
});
