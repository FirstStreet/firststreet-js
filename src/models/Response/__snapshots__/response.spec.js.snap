// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock should initialize a base model 1`] = `
Response {
  "currentData": Array [
    Object {
      "floodData": Array [
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.17557,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.09307,
            },
          ],
          "year": 2018,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.18321,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.09554,
            },
          ],
          "year": 2023,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.19847,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.09909,
            },
          ],
          "year": 2028,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.20992,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.10318,
            },
          ],
          "year": 2033,
        },
      ],
      "floodID": "c1",
    },
    Object {
      "floodData": Array [
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 11,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 1,
            },
          ],
          "year": 2018,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 11,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 1,
            },
          ],
          "year": 2023,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 11,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 1,
            },
          ],
          "year": 2028,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 11,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 1,
            },
          ],
          "year": 2033,
        },
      ],
      "floodID": "c5",
    },
    Object {
      "floodData": Array [
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 7,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.98092,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.90534,
            },
          ],
          "year": 2018,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 7,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.98092,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.9098,
            },
          ],
          "year": 2023,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 7,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.98092,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.91659,
            },
          ],
          "year": 2028,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 7,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 1,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.98473,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.92386,
            },
          ],
          "year": 2033,
        },
      ],
      "floodID": "c3",
    },
  ],
  "data": Object {
    "FSID": 450350219571,
    "results": Array [
      Object {
        "floodData": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.17557,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09307,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.18321,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09554,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.19847,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09909,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.20992,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.10318,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c1",
      },
      Object {
        "floodData": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c5",
      },
      Object {
        "floodData": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.90534,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.9098,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.91659,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98473,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.92386,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c3",
      },
    ],
  },
}
`;
